dnf/cli/commands/group.py

```diff
@@ -177,7 +177,7 @@ def _list(self, userlist):
         def _out_grp(sect, group):
             if not done:
                 print(sect)
-            msg = '   %s' % group.ui_name
+            msg = '   %s' % (group.ui_name if group.ui_name is not None else _("<name-unset>"))
```

Comment:
I thing that the solution will return in additional reports. The additional problem will be in usage of list command. The output of list command (even when the group without the name is installed) will not work with any other dnf group command like "dnf group install, upgrade, info".

Reply:
Also there is no warning or an error that will inform users that something is wrong.

Reply:
> I thing that the solution will return in additional reports. The additional problem will be in usage of list command. The output of list command (even when the group without the name is installed) will not work with any other dnf group command like "dnf group install, upgrade, info".

It is not meant to work. The commands are designed to work with group names as identifiers (which is a bad design IMHO). Therefore the group names should always be set. If they are not, it is an error state that needs to fixed. We shouldn't try to hide it and work around it instead.

> Also there is no warning or an error that will inform users that something is wrong.

Right.  FWIW I don't find the warning strictly necessary, but if you want I'll add it.

I'm not sure how to do this though. If we should print a warning, the warning should go into the code loading the comps data, I don't want to duplicate the warning in various places the name attribute is used. But then it will be printing the warning on every command? Should I do that?

Reply:
> > I thing that the solution will return in additional reports. The additional problem will be in usage of list command. The output of list command (even when the group without the name is installed) will not work with any other dnf group command like "dnf group install, upgrade, info".
> 
> It is not meant to work. The commands are designed to work with group names as identifiers (which is a bad design IMHO). Therefore the group names should always be set. If they are not, it is an error state that needs to fixed. We shouldn't try to hide it and work around it instead.

Incorrect, commands are designed to work with group names and group ids.


> 
> > Also there is no warning or an error that will inform users that something is wrong.
> 
> Right. FWIW I don't find the warning strictly necessary, but if you want I'll add it.
> 
> I'm not sure how to do this though. If we should print a warning, the warning should go into the code loading the comps data, I don't want to duplicate the warning in various places the name attribute is used. But then it will be printing the warning on every command? Should I do that?

Yes, that would be a better approach than the patch.

Please use the same approach like when in .repo file is missing the name key. Then it uses repoid as a name and generates a warning. A user gets fully working dnf and also possibility to report a problem in metadata. Additionally it unify behaviour with other parts of dnf.

Reply:
> > It is not meant to work. The commands are designed to work with group names as identifiers (which is a bad design IMHO). Therefore the group names should always be set. If they are not, it is an error state that needs to fixed. We shouldn't try to hide it and work around it instead.
> 
> Incorrect, commands are designed to work with group names and group ids.

Yes, they work with ids in addition to names. That doesn't make my statement incorrect and the names should still work.

> > Right. FWIW I don't find the warning strictly necessary, but if you want I'll add it.
> > I'm not sure how to do this though. If we should print a warning, the warning should go into the code loading the comps data, I don't want to duplicate the warning in various places the name attribute is used. But then it will be printing the warning on every command? Should I do that?
> 
> Yes, that would be a better approach than the patch.
> 
> Please use the same approach like when in .repo file is missing the name key. Then it uses repoid as a name and generates a warning. A user gets fully working dnf and also possibility to report a problem in metadata. Additionally it unify behaviour with other parts of dnf.

No, I want to keep my current approach. I don't want to arbitrarily change the attribute that is printed out on the output based on whether the name is set. That is just inconsistent and unreliable. It should always be clear what is what in the output, with this approach you would never know whether you're working with a name or with an id (except for the warning, but that is printed on stderr and would have no effect in a script).

It seems you're considering a case where a user would pipe the output of `dnf group list` to e.g. `dnf group install` and want to make it work even if the names aren't present. I find this usage unlikely (you usually know what particular groups to install, why would you get them from the list).

Instead, consider a case where a user writes a script to get all group names using the `dnf group list` command and stores them somewhere for a later use. With your approach, some of the names wouldn't be names, they would be ids. Sure, they would probably work for the most part, but they are still a different attribute and the user has no way to distinguish them programatically anymore. That is a bad design.

Reply:
I believe that the discussion is not leading anywhere. String that you want to provide as a group name  _("<name-unset>") is not group name anyway.

The string that will be provided in output is unpredictable because it depends on translation and could be changed any time. Also your change does not provide any information on stderr that something is wrong.

Reply:
> I believe that the discussion is not leading anywhere.

Yes... :slightly_frowning_face: 

> String that you want to provide as a group name _("") is not group name anyway.

It is not indeed, since the name doesn't exist. It represents the non-existent name.

> The string that will be provided in output is unpredictable because it depends on translation and could be changed any time.

That's part of the current, IMHO broken design. Nevertheless, it is the name. What if you're listing and storing those to show them to a user and want to have nice translated names.

> Also your change does not provide any information on stderr that something is wrong.

We were just discussing that I can add a warning to `stderr` if you want.

Reply:
Does it make sense to extend <name-unset> output by group id?
In current situation when i run `dnf group list` I'm getting information that some unnamed group exists, but I do not have means how to find out which one it actually is. At least warning in `stderr` containing the group might be enough.

Reply:
> Does it make sense to extend `<name-unset>` output by group id?
> In current situation when i run `dnf group list` I'm getting information that some unnamed group exists, but I do not have means how to find out which one it actually is.

You can use the `--ids` switch to show the ids...

> At least warning in `stderr` containing the group id might be enough.

Yeah... I can add that.

Reply:
Given that name is not a required field in group element we should not produce warnings when it is missing.